### PR TITLE
RTC implementation of pcf8563

### PR DESCRIPTION
**Pull request by KoenBreeman**
# What does this implement/fix?

Implementation of the feature request for PCF8563.
https://github.com/esphome/feature-requests/issues/2173

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Other

**Related issue or feature (if applicable):** fixes <link to issue>
https://github.com/esphome/feature-requests/issues/2173

**Pull request in [esphome-docs](https://github.com/esphome/esphome-docs) with documentation (if applicable):** esphome/esphome-docs#2868

## Test Environment

- [X] ESP32
- [ ] ESP32 IDF
- [ ] ESP8266
- [ ] RP2040

## Example entry for `config.yaml`:

```yaml
# Example config.yaml
external_components:
  - source: github://pr#4998
    components: pcf8563

time:
  - platform: pcf8563

on_...:
  - pcf8563.read_time

  # in case you need to specify the PCF8563 id
  - pcf8563.read_time:
      id: pcf8563_time

```

## Checklist:
  - [x] The code change is tested and works locally.
  - [x] Tests have been added to verify that the new code works (under `tests/` folder).

If user exposed functionality or configuration variables are added/changed:
  - [x] Documentation added/updated in [esphome-docs](https://github.com/esphome/esphome-docs). 
  https://github.com/KoenBreeman/ESPHome_doc_fork/tree/PCF8563

### CODE REVIEW

**Comment by KoenBreeman** [comment]
Checked at pull request and everything works

File: esphome/components/pcf8563/time.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+)
+WriteAction = pcf8563_ns.class_("WriteAction", automation.Action)
+ReadAction = pcf8563_ns.class_("ReadAction", automation.Action)
+
+
+CONFIG_SCHEMA = time.TIME_SCHEMA.extend(
+    {
+        cv.GenerateID(): cv.declare_id(pcf8563Component),
+    }
+).extend(i2c.i2c_device_schema(CONF_I2C_ADDR))
```
Comment (jesserockz) [inline esphome/components/pcf8563/time.py]:
```suggestion
).extend(i2c.i2c_device_schema(0xA3))
```

File: esphome/components/pcf8563/time.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+import esphome.codegen as cg
+from esphome import automation
+from esphome.components import i2c, time
+from esphome.const import CONF_ID
+
+CODEOWNERS = ["@KoenBreeman"]
+
+DEPENDENCIES = ["i2c"]
+
+CONF_I2C_ADDR = 0xA3
```
Comment (jesserockz) [inline esphome/components/pcf8563/time.py]: We dont static store addresses in variables
```suggestion
```

**Review by jesserockz — DISMISSED** [review]
Mostly looks good, one minor change

**Review by jesserockz — APPROVED** [review]

**Pull request merged**